lib/core/sync_job_runner.rb

```diff
@@ -83,10 +85,12 @@ def do_sync!
         Utility::Logger.info("Upserted #{@status[:indexed_document_count]} documents into #{@connector_settings.index_name}.")
         Utility::Logger.info("Deleted #{@status[:deleted_document_count]} documents into #{@connector_settings.index_name}.")
 
+        @status[:error] = 'Sync thread didn\'t finish execution. Check connector logs for more details.' unless @sync_finished
```

Comment:
Looks like this will trigger also in cases when an error happens, because @sync_finished is never set to `true` in `rescue` block.

Curiously it's not caught by the test, I think it's worth to add a test for it:
1. When an StandardError is caught, then this error is properly put to `status[:error]`
2. When no error is caught, but sync is not finished, only then set the error to the message on this line